File: hex_digit.js
const {PiBBP} = require("./digit_extraction/digit_extraction");

let N = BigInt(1);
if (process.argv.length >= 3) {
    N = BigInt(process.argv[2]);
}

const hex2char = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f"]
console.log(hex2char[PiBBP(n)]);

File: hex_digit.go
package main

import (
	"flag"
	"fmt"
	"github.com/khanhhhh/pi_digit_extraction/digit_extraction"
	"math/big"
)

func main() {
	var hex2char = [16]string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f"}
	flag.Parse()
	args := flag.Args()
	var n = big.NewInt(1)
	if len(args) >= 1 {
		temp, ok := (&big.Int{}).SetString(args[0], 10)
		if ok {
			n = temp
		}
	}
	fmt.Println(hex2char[digit_extraction.PiBPP(n)])
}
